Provider/src/main/java/com/spotify/confidence/EventSenderUploader.kt

```diff
@@ -0,0 +1,74 @@
+package com.spotify.confidence
+
+import com.spotify.confidence.client.ConfidenceValueMap
+import com.spotify.confidence.client.await
+import com.spotify.confidence.client.serializers.DateSerializer
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.withContext
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.encodeToString
+import kotlinx.serialization.json.Json
+import okhttp3.Headers
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.RequestBody.Companion.toRequestBody
+import java.util.Date
+
+internal interface EventSenderUploader {
+    suspend fun upload(events: EventBatch): Boolean
+}
+
+@Serializable
+internal data class EventBatch(
```

Comment:
We should check with the backend but I think we should also be able to send the `sdk` data here.
From java:
```
Sdk.newBuilder()
            .setId(SdkId.SDK_ID_JAVA_CONFIDENCE)
            .setVersion(SdkUtils.getSdkVersion())
            .build();
```